src/coreclr/jit/importer.cpp

```diff
@@ -13064,27 +13064,36 @@ void Compiler::impImportBlockCode(BasicBlock* block)
                 op2 = impPopStack().val;
                 op1 = impPopStack().val;
 
+                // Recognize the IL idiom of CGT_UN(op1, 0) and normalize
+                // it so that downstream optimizations don't have to.
+                if ((opcode == CEE_CGT_UN) && op2->IsIntegralConst(0))
```

Comment:
can you please check the case where op1 is zero (reversed) here - does it produce any diffs?

Reply:
I don't think that's necessary: `cgt.un(0, op)` is always `false`, and `op1 = gtFoldExpr(op1)` below [takes care of that](https://github.com/dotnet/runtime/blob/d87b78342c5d88741eb1f194b4c9968baf192ddc/src/coreclr/jit/gentree.cpp#L13377).

Reply:
@SingleAccretion no, I meant `clt.un(0, op)` the same pattern but reversed. Also, `cle.un(op, 0)` -- the same transformation in morph does that https://github.com/dotnet/runtime/blob/main/src/coreclr/jit/morph.cpp#L13917-L13918

Reply:
I see, thank you for the clarification.

So I did add the `clt.un(0, op)` handling here, and the diffs were in about 10 methods across all collections, mostly positive as one would expect (amounting to ~150 bytes), but there was was one regression that got me thinking a bit. The regression is because `GT_UN(x, 0)` (reversed from `LT(0, x)`) allows assertion prop to eliminate a range check against zero, while `NE(x, 0)` doesn't have the same effect. Perhaps we should not be normalizing this idiom after all...

I will say that if it is ok, I would prefer to leave the case here as is. The long-term plan (well, plan for the PR after the next one) is to just fix the ordering problem in morph.

Reply:
I actually tried exactly the same opt (to recognize GT_NE early in the importer) a long time ago but there were lots of range-checks regressions and I gave up so I guess those were fixed since then (by you? 🙂)

Reply:
By @nathan-moore's #40180 would be my guess :).

Reply:
Yea, that should handle ==,!= 0 in range check. It's wierd that you saw regressions though since it got changed to != in morph before anything that should remove bound checks ran.  ¯\\\_(ツ)_/¯